inst/tests/tests.Rraw

```diff
@@ -16796,37 +16795,24 @@ if (test_xts) {
+  test(2108.36, first(xt, n=2L), xt_first2, output="using xts::first")
 }
-test(2108.41, last(x), 3L, output="using 'x[[length(x)]]': !is.xts(x) & !nargs>1 & is.null(dim(x))")
-test(2108.42, last(y), y[3L], output="using 'x[[length(x)]]': !is.xts(x) & !nargs>1 & is.null(dim(x))")
-test(2108.51, first(x), 1L, output="using 'x[[1L]]': !is.xts(x) & !nargs>1 & is.null(dim(x))")
-test(2108.52, first(y), y[1L], output="using 'x[[1L]]': !is.xts(x) & !nargs>1 & is.null(dim(x))")
-test(2108.61, last(df), structure(list(a=2L, b=2L), row.names=2L, class="data.frame"), output="using 'x[nrow(x),]': !is.xts(x) & !nargs>1 & is.data.frame(x)")
-test(2108.62, last(dt), data.table(a=2L, b=2L), output="using 'x[nrow(x),]': !is.xts(x) & !nargs>1 & is.data.frame(x)")
-test(2108.71, first(df), structure(list(a=1L, b=3L), row.names=1L, class="data.frame"), output="using 'x[1L,]': !is.xts(x) & !nargs>1 & is.data.frame(x)")
-test(2108.72, first(dt), data.table(a=1L, b=3L), output="using 'x[1L,]': !is.xts(x) & !nargs>1 & is.data.frame(x)")
+test(2108.41, last(x), 3L)
+test(2108.42, last(y), y[3L])
+test(2108.51, first(x), 1L)
+test(2108.52, first(y), y[1L])
+test(2108.610, last(df), structure(list(a=2L, b=2L), row.names=2L, class="data.frame"))
+test(2108.611, last(as.data.table(df), na.rm=TRUE), data.table(a=2L,b=2L))
+test(2108.62, last(dt), data.table(a=2L, b=2L))
+test(2108.710, first(df), structure(list(a=1L, b=3L), row.names=1L, class="data.frame"))
+test(2108.711, first(as.data.table(df), na.rm=TRUE), data.table(a=1L, b=3L))
+test(2108.72, first(dt), data.table(a=1L, b=3L))
 # matrix/array utils::tail behavior is likely to change in future R, Michael is more in the topic
-test(2108.81, last(mx), structure(c(3L, 6L, 9L), .Dim = c(1L, 3L), .Dimnames = list("[3,]", NULL)), output="using utils::tail: !is.xts(x) & !nargs>1 & !is.null(dim(x)) & !is.data.frame(x)")
-expected = if (base::getRversion() < "3.7.0") 27L else structure(c(3L, 6L, 9L, 12L, 15L, 18L, 21L, 24L, 27L), .Dim = c(1L, 3L, 3L), .Dimnames = list("[3,]", NULL, NULL)) #4127
-test(2108.82, last(ar), expected, output="using utils::tail: !is.xts(x) & !nargs>1 & !is.null(dim(x)) & !is.data.frame(x)")
-test(2108.91, first(mx), structure(c(1L, 4L, 7L), .Dim = c(1L, 3L)), output="using utils::head: !is.xts(x) & !nargs>1 & !is.null(dim(x)) & !is.data.frame(x)")
-expected = if (base::getRversion() < "3.7.0") 1L else structure(c(1L, 4L, 7L, 10L, 13L, 16L, 19L, 22L, 25L), .Dim = c(1L, 3L, 3L)) #4127
-test(2108.92, first(ar), expected, output="using utils::head: !is.xts(x) & !nargs>1 & !is.null(dim(x)) & !is.data.frame(x)")
+test(2108.81, last(mx), structure(c(3L, 6L, 9L), .Dim = c(1L, 3L), .Dimnames = list("[3,]", NULL)))
+test(2108.82, last(ar), if (base::getRversion()<"3.7.0") 27L else structure(c(3L, 6L, 9L, 12L, 15L, 18L, 21L, 24L, 27L), .Dim = c(1L, 3L, 3L), .Dimnames = list("[3,]", NULL, NULL)))  # utils:tail, #4127
+test(2108.91, first(mx), structure(c(1L, 4L, 7L), .Dim = c(1L, 3L)))
```

Comment:
the test using structure() is a bit hard to follow. either `rbind(1,4,7)` or `matrix(1,4,7, ncol = 3)` would be easier to follow

Reply:
`test(2108.91, first(mx), matrix(c(1L,4L,7L), ncol=3))` would do.

Reply:
done